File: Year I/Semester I/Programarea Algorimilor/Materiale/2022 - 2023/Laboratoare/Grupa 152/Rezolvari/Laborator 06/05.py
def ex5(s):
    for ch in "!?.":
        s = s.replace(ch, "$")
    return f"Sirul este alcatuit din {len(s.split('$'))} propozitii. Cea mai lunga propozitie are {max([len(x) for x in s.split('$')])} caractere"


s = input()
print(ex5(s))

File: Year I/Semester I/Logica Matematica Si Computationala/Materiale/2022 - 2023/Laboratoare/Rezolvari/Laborator 03/Exercitii/05.pl
split(_, [], [], []).
split(X, [H|T], [H|A], B):-
    H < X, split(X, T, A, B).
split(X, [H|T], A, [H|B]):-
    H >= X, split(X, T, A, B).

quicksort([],[]).
quicksort([H|T], L) :-
    split(H, T, A, B), quicksort(A, M), quicksort(B, N), append(M, [H|N], L).

